dagstore_control.go

```diff
@@ -297,8 +299,12 @@ func (d *DAGStore) control() {
 
 			d.lk.Lock()
 			delete(d.shards, s.key)
+
+			// Delete the entry directly from the datastore as persist does not implement deletion
+			// Persist should not make any changes for the control_loop runs for deletion
+			err = d.store.Delete(d.ctx, datastore.NewKey(s.key.String()))
```

Comment:
Currently the dagstore assumes that operations against the datastore will succeed, and just logs an error if they fail.
So I'd suggest we continue to follow that pattern here. Let's move the call to `s.store.Delete` to the if / else block on line 318